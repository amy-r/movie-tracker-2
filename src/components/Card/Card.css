.Card {
  background-color: #333;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 280px;
  margin: 20px;
  width: 200px;
  box-shadow: 0px 0px 5px 5px  #4a4949;
}

.Card:hover {
    box-shadow: 1px 1px 10px 5px #ccc;
}


.favorite {
  border: 2px solid white;
  background-color: ghostwhite;
  color: black;
}

h2 {
  font-weight: 400;
  font-family: 'Roboto', sans-serif;
  margin: 5px;
}

img {
  width: 100%;
  min-height: 100%;
}

button {
  background-color: red;
  color: white;
  display: inline-block;
  border: none;
  border-radius: 20px;
  font-size: 14px;
  line-height: 0px;
  height: 30px;
  width: 90px;
  padding: 10px 15px;
  margin: -30px 5px 10px 5px;
  z-index: 1;
}

button:hover {
  background-color: purple;
  box-shadow: 2px 2px 5px black;
}