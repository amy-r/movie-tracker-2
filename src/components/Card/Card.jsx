import React from 'react';
import './Card.css';

const Card = ({ title, poster }) => {
  return (
    <div className="Card">
      <div className='favorite' >
      </div>
      <img height='250' width='200' src={`https://image.tmdb.org/t/p/w500/${poster}`} alt={`${title} movie poster`}/>
    </div>
  );
};

export default Card;
