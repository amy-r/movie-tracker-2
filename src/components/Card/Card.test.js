import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

describe('CARD', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Card movie={{title:'Gone With The Wind'}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
