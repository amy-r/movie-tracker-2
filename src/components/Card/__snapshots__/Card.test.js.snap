// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CARD should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    movie={
        Object {
            "title": "Gone With The Wind",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Gone With The Wind movie poster"
          src="https://image.tmdb.org/t/p/w500/undefined"
/>,
        <button
          className="card-button"
          id={undefined}
          onClick={undefined}
>
          Favorite
</button>,
      ],
      "className": "Card undefined",
      "id": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Gone With The Wind movie poster",
          "src": "https://image.tmdb.org/t/p/w500/undefined",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorite",
          "className": "card-button",
          "id": undefined,
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Favorite",
        "type": "button",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Gone With The Wind movie poster"
            src="https://image.tmdb.org/t/p/w500/undefined"
/>,
          <button
            className="card-button"
            id={undefined}
            onClick={undefined}
>
            Favorite
</button>,
        ],
        "className": "Card undefined",
        "id": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Gone With The Wind movie poster",
            "src": "https://image.tmdb.org/t/p/w500/undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "className": "card-button",
            "id": undefined,
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "button",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
